examples/test_util/ERst.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+   Copyright 2019 Equinor ASA.
+
+   This file is part of the Open Porous Media project (OPM).
+
+   OPM is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   OPM is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+   */
+
+#include "ERst.hpp"
+
+#include <string>
+#include <string.h>
+#include <sstream>
+#include <iterator>
+#include <iomanip>
+#include <algorithm>
+
+
+ERst::ERst(const std::string& filename) : EclFile(filename)
+{
+    loadData("SEQNUM");
+
+    std::vector<int> firstIndex;
+
+    for (size_t i = 0;  i < array_name.size(); i++) {
+        if (array_name[i] == "SEQNUM") {
+            auto seqn = get<int>(i);
+            seqnum.push_back(seqn[0]);
+            firstIndex.push_back(i);
+        }
+    }
+
+
+    for (size_t i = 0; i < seqnum.size(); i++) {
+        std::pair<int,int> range;
+        range.first = firstIndex[i];
+
+        if (i != seqnum.size() - 1) {
+            range.second = firstIndex[i+1];
+        } else {
+            range.second = array_name.size();
+        }
+
+        arrIndexRange[seqnum[i]] = range;
+    }
+
+    nReports = seqnum.size();
+
+    for (int i = 0; i < nReports; i++) {
+        reportLoaded[seqnum[i]] = false;
```

Comment:
At a glance this looks like an out-of-bounds access in `reportLoaded`?

Reply:
no, it's a map. operator [] handles insertion.

Reply:
Ahhh - ok :)